### ISSUE TITLE
neural network training methods should be able to stop at convergence

### ISSUE THREAD

**Issue by sbenthall** [scikit-agent/scikit-agent#92]
The neural network training methods currently run for a fixed number of epochs, rather than stopping at convergence.

https://github.com/scikit-agent/scikit-agent/blob/main/src/skagent/ann.py#L300

As a new feature, we could have these methods stop at the point where the loss function flattens out.
